src/main/java/appeng/recipes/game/WirelessTerminalRecipe.java

```diff
@@ -0,0 +1,159 @@
+{
+
+    private final IComparableDefinition terminal;
+    private final ItemStack waterBucket;
+
+    private static final ArrayList<String> dyes = new ArrayList<String>();
+    static
+    {
+        dyes.add( "dyeBlack" );
+        dyes.add( "dyeBlue" );
+        dyes.add( "dyeBrown" );
+        dyes.add( "dyeCyan" );
+        dyes.add( "dyeGray" );
+        dyes.add( "dyeGreen" );
+        dyes.add( "dyeLightBlue" );
+        dyes.add( "dyeLightGray" );
+        dyes.add( "dyeLime" );
+        dyes.add( "dyeMagenta" );
+        dyes.add( "dyeOrange" );
+        dyes.add( "dyePink" );
+        dyes.add( "dyePurple" );
+        dyes.add( "dyeRed" );
+        dyes.add( "dyeWhite" );
+        dyes.add( "dyeYellow" );
+    }
+
+    public WirelessTerminalRecipe()
+    {
+        final IDefinitions definitions = AEApi.instance().definitions();
+        terminal = definitions.items().wirelessTerminal();
+        waterBucket = new ItemStack( Items.water_bucket, 1 );
+    }
+
+    @Override
+    public boolean matches( InventoryCrafting inv, World w )
+    {
+        return this.getOutput( inv, false ) != null;
+    }
+
+    @Nullable
+    private ItemStack getOutput( final IInventory inv, final boolean createTerminal )
+    {
+        ItemStack inputTerminal = null;
+        String color = null;
+        int numIngredients = 0;
+        for( int i = 0; i < 9; i++ )
+        {
+            //If the slot is empty, ignore it
+            if( inv.getStackInSlot( i ) != null )
+            {
+                numIngredients++;
+                //if it contains a terminal, return that terminal
+                if( this.terminal.isSameAs( inv.getStackInSlot( i ) ) )
+                {
+                    inputTerminal = inv.getStackInSlot( i );
+                }
+                else
+                {
+                    //check if the other item is a dye
+                    color = getColorFromItem( inv.getStackInSlot( i ) );
+                }
+            }
+
+        }
+
+        //Check if both a terminal and a dye were found
+        if( inputTerminal==null || color == null || numIngredients!=2)
+        {
+            return null;
+        }
+
+        ItemStack output = ItemStack.copyItemStack( inputTerminal );
+        if( createTerminal )
+        {
+            NBTTagCompound tag = output.getTagCompound();
+            tag.setString( "color", color );
+        }
+
+        return output;
+    }
+
+    private String getColorFromItem(ItemStack potentialColor)
+    {
+        //check if the "color" is water, in which case return fluix
+        if( ItemStack.areItemStacksEqual( waterBucket, potentialColor ) )
+        {
+            return "fluix";
+        }
+
+        //check through the ore dictionary to see if the item is a dye
+        for( String dye : dyes )
+        {
+            ArrayList<ItemStack> dyeItems = OreDictionary.getOres( dye );
+            for( ItemStack itm : dyeItems )
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Local variable 'itm' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)